fendermint/vm/interpreter/src/fvm/mod.rs

```diff
@@ -23,18 +24,42 @@ pub use fendermint_vm_message::query::FvmQuery;
 pub use genesis::FvmGenesisOutput;
 pub use query::FvmQueryRet;
 
+pub use self::broadcast::Broadcaster;
 use self::state::ipc::GatewayCaller;
 
 pub type FvmMessage = fvm_shared::message::Message;
 
+#[derive(Clone)]
+pub struct ValidatorContext<C> {
+    /// The secret key the validator uses to produce blocks.
+    secret_key: SecretKey,
```

Comment:
Out of a curiosity, how are we passing the `secret_key` to the context?

Reply:
If the settings are pointing at validator key file, we read it and create a context from it here: https://github.com/consensus-shipyard/fendermint/pull/276/files#diff-6d8c711ef368a10a52b47f7702c1eff949ea4bac181370ffae4e8f49f3e1f9a2R46